ui/components/app/alerts/unconnected-account-alert/unconnected-account-alert.js

```diff
@@ -43,16 +42,8 @@ const UnconnectedAccountAlert = () => {
     return account;
   });
   const origin = useSelector(getOriginOfCurrentTab);
-  const selectedIdentity = useSelector(getSelectedIdentity);
-  // Temporary fix until https://github.com/MetaMask/metamask-extension/pull/21553
-  const selectedIdentityWithName = {
-    ...selectedIdentity,
-    name: internalAccounts.find(
-      (internalAccount) =>
-        internalAccount.address === selectedIdentity?.address,
-    )?.metadata.name,
-  };
-  const selectedAddress = useSelector(getSelectedAddress);
+  const account = useSelector(getSelectedInternalAccount);
+  const { address: selectedAddress } = account;
```

Comment:
I do not think we should remove the logic for the account names as it results in different names in the account list then in the send flow. A better solution would ideally be to export the account name metadata from this selector.

Reply:
retested with the later commits and this is no longer relevant.